pkg/cmd/repo/fork/fork.go

```diff
@@ -0,0 +1,292 @@
+package fork
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/cli/cli/api"
+	"github.com/cli/cli/context"
+	"github.com/cli/cli/git"
+	"github.com/cli/cli/internal/config"
+	"github.com/cli/cli/internal/ghrepo"
+	"github.com/cli/cli/internal/run"
+	"github.com/cli/cli/pkg/cmdutil"
+	"github.com/cli/cli/pkg/iostreams"
+	"github.com/cli/cli/pkg/prompt"
+	"github.com/cli/cli/utils"
+	"github.com/spf13/cobra"
+)
+
+type ForkOptions struct {
+	HttpClient func() (*http.Client, error)
+	Config     func() (config.Config, error)
+	IO         *iostreams.IOStreams
+	BaseRepo   func() (ghrepo.Interface, error)
+	Remotes    func() (context.Remotes, error)
+	Since      func(t time.Time) time.Duration
+
+	Repository   string
+	Clone        bool
+	Remote       bool
+	PromptClone  bool
+	PromptRemote bool
+}
+
+// TODO decide if this should be injected via a factory.
+var Since = func(t time.Time) time.Duration {
+	return time.Since(t)
+}
+
+func NewCmdFork(f *cmdutil.Factory, runF func(*ForkOptions) error) *cobra.Command {
+	opts := &ForkOptions{
+		IO:         f.IOStreams,
+		HttpClient: f.HttpClient,
+		Config:     f.Config,
+		BaseRepo:   f.BaseRepo,
+		Remotes:    f.Remotes,
+	}
+
+	cmd := &cobra.Command{
+		Use:   "fork [<repository>]",
+		Args:  cobra.MaximumNArgs(1),
+		Short: "Create a fork of a repository",
+		Long: `Create a fork of a repository.
+
+With no argument, creates a fork of the current repository. Otherwise, forks the specified repository.`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			promptOk := opts.IO.IsStdoutTTY() && opts.IO.IsStdinTTY()
+			if len(args) > 0 {
+				opts.Repository = args[0]
+			}
+
+			if cmd.Flags().Changed("clone") && cmd.Flags().Changed("no-clone") {
+				return errors.New("--clone and --no-clone are mutually excusive")
+			}
+
+			if cmd.Flags().Changed("remote") && cmd.Flags().Changed("no-remote") {
+				return errors.New("--remote and --no-remote are mutually excusive")
+			}
+
+			if promptOk && (!cmd.Flags().Changed("clone") && !cmd.Flags().Changed("no-clone")) {
+				opts.PromptClone = true
+			}
+
+			if promptOk && (!cmd.Flags().Changed("remote") && !cmd.Flags().Changed("no-remote")) {
+				opts.PromptRemote = true
+			}
+
+			if cmd.Flags().Changed("no-clone") {
```

Comment:
so...

I think I have been confused. I honestly didn't realize that cobra bool flags could be passed like `--clone=false` and I think instead of opening up the `-(no)-clone` can of worms I'd rather just do that for now.

The initial problem was that I thought cobra couldn't distinguish between:

- no flags specified (ie, prompt)
- desire is to set flag to false (which i thought could only be done by omitting the flag)

but since `--clone=false` works, I *can* identify the "desire for false" vs. the "no flags specified" case :facepalm: I'm going to retool for that.

Reply:
I switched to --clone/--remote as bools and things are fine :+1: